### ISSUE TITLE
Node: selenium-webdriver: UnsupportedOperationError when taking screenshot of elemet with a remote server

### ISSUE THREAD

**Issue by audioXD** [SeleniumHQ/selenium#5136]
## Meta -
OS:  linux (arch)

Selenium Server Version:  3.7.1
selenium-webdriver Version: 3.0.8

Browser:  firefox
Browser Version:  Mozilla Firefox 57.0
Geckodriver Driver:  0.19.0

## Expected Behavior -
To take screenshot of element.

## Actual Behavior -
Throws UnsupportedOperationError:

## Steps to reproduce -
In node:
`let driver = new Builder()
  .forBrowser("firefox")
  .usingServer('http://localhost:4444/wd/hub')
  .build();`

and a selenium server sunning
`java -jar selenium-server-standalone-3.7.1.jar`
I took a screenshot of a element with
`element.takeScreenshot(true)`

and got: 
`node:6930) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): UnsupportedOperationError: GET /session/4cfba22f-cd2f-4e3a-8547-0c7a8eeb3624/element/e3226585-6fe0-409c-a431-47fb9fc8c308/screenshot
Build info: version: '3.7.1', revision: '8a0099a', time: '2017-11-06T21:07:36.161Z'
System info: host: 'audioxd', ip: '127.0.0.1', os.name: 'Linux', os.arch: 'amd64', os.version: '4.13.12-1-ARCH', java.version: '1.8.0_144'
Driver info: driver.version: unknown.`

**NOTE: If I remove the usingServer option it works.**

**Comment by jleyba** [comment]
Looks like the java server doesn't support the element screenshot command.

**Comment by audioXD** [comment]
And why doesn't it. Thats what I'm curious about.

**Comment by jleyba** [comment]
Because it hasn't been implemented?

**Comment by ryancat** [comment]
I got same error:

```
UnsupportedOperationError: GET /session/XXXXXXXX-XXXX-XXXX-XXXX-XXXX188ccd15/element/1/screenshot
Build info: version: '3.4.0', revision: 'unknown', time: 'unknown'
System info: host: 'WIN-R1JBAO1CNPS', ip: '172.20.42.188', os.name: 'Windows Server 2008 R2', os.arch: 'x86', os.version: '6.1', java.version: '1.8.0_91'
Driver info: driver.version: unknown
      at Object.checkLegacyResponse (node_modules/selenium-webdriver/lib/error.js:585:15)
      at parseHttpResponse (node_modules/selenium-webdriver/lib/http.js:533:13)
      at Executor.execute (node_modules/selenium-webdriver/lib/http.js:468:26)
      at process._tickCallback (internal/process/next_tick.js:109:7)
```

**Comment by ryancat** [comment]
Since Webdriverio works on taking screenshot for element, I figured they did it by applying css translate on html, and resize the window size to the actual element size to take a screenshot of the whole browser. That seems working and it's pretty smart.

See https://github.com/zinserjan/wdio-screenshot/blob/e903b946a7725a60b6f3cbdc13caae567fa1570e/src/modules/makeAreaScreenshot.js

**Comment by jimevans** [comment]
@ryancat Getting an element screenshot hasn’t been implemented by the IE driver yet. You’ll receive the same result with any language binding that calls that http end point with the IE driver.

**Comment by barancev** [comment]
There was a bug in element screenshot command that was fixed in version 3.9, please retest the issue in the latest version (taht is 3.11 at the moment). And, yes, you have to use a driver that supports this command (Firefox only at the moment)

**Comment by ryancat** [comment]
I followed the link https://github.com/zinserjan/wdio-screenshot/blob/e903b946a7725a60b6f3cbdc13caae567fa1570e/src/modules/makeAreaScreenshot.js and found a work around by cropping the whole screenshot to get the image for just that element.

**Comment by barancev** [comment]
Closing the issue, it's supposed to be fixed in version 3.11.
